web3.md

```diff
@@ -1255,28 +1256,44 @@ Transaction Execution
     syntax KItem ::= "#validateTx" Int
  // ----------------------------------
     rule <k> #validateTx TXID => . ... </k>
-         <statusCode> ( _ => EVMC_OUT_OF_GAS) </statusCode>
+         <statusCode> _ => #if BAL <Int GLIMIT *Int GPRICE #then EVMC_BALANCE_UNDERFLOW #else EVMC_OUT_OF_GAS #fi </statusCode>
          <schedule> SCHED </schedule>
+         <origin> ACCTFROM </origin>
+         <account>
+           <acctID> ACCTFROM </acctID>
+           <balance> BAL </balance>
+           ...
+         </account>
          <message>
            <msgID>      TXID   </msgID>
+           <txGasPrice> GPRICE </txGasPrice>
            <txGasLimit> GLIMIT </txGasLimit>
            <data>       DATA   </data>
            <to>         ACCTTO </to>
            ...
          </message>
-      requires ( GLIMIT -Int G0(SCHED, DATA, (ACCTTO ==K .Account)) ) <Int 0
+      requires GLIMIT -Int G0(SCHED, DATA, (ACCTTO ==K .Account)) <Int 0
+        orBool BAL <Int GLIMIT *Int GPRICE
```

Comment:
I didn't catch this before, but we should likely do instead `<k> #validateTx TXID => #if BAL <Int GLIMIT *Int GPRICE #then #end EVMC_BALANCE_UNDERFLOW #else #end EVMC_OUT_OF_GAS #fi ... </k>`

`#end ...` is used to halt execution and signal an exception.

Reply:
That will leave a `#halt` at the top of the cell which I'm pretty sure will cause a problem, as it will do `<k> #halt ~> (_:Int => .) ... </k>`, and we have `TXID:Int` in the k cell for `#eth_sendTransaction_final`.